src/index.js

```diff
@@ -18,14 +18,23 @@ import { createExecuteCommand } from "./core";
 import createLogger from "./core/createLogger";
 import createLogController from "./core/createLogController";
 import { injectStorage } from "./utils";
+import { arrayOf, objectOf, string } from "./utils/validation";
 
 const { console } = window;
 const createNamespacedStorage = injectStorage(window);
-// set this up as a function so that monitors can be added at anytime
-// eslint-disable-next-line no-underscore-dangle
-const defaultGetMonitors = () => window.__alloyMonitors || [];
 
-export const createInstance = ({ name, getMonitors = defaultGetMonitors }) => {
+export const createInstance = options => {
+  const eventOptionsValidator = objectOf({
+    name: string().required(),
+    monitors: arrayOf(objectOf({}))
```

Comment:
you could do this:

monitors: arrayOf(objectOf({})).default([])

Even name could probably default to "alloy".